debian/cuttlefish-common.cuttlefish-host-orchestrator.init

```diff
@@ -0,0 +1,118 @@
+#!/bin/sh
+#
+### BEGIN INIT INFO
+# Provides: cuttlefish-host-orchestrator
+# Required-Start: $network $remote_fs
+# Required-Stop: $network $remote_fs
+# Default-Start:
+# Default-Stop: 0 1 2 3 4 5 6
+# Short-Description: Cuttlefish Host Orchestrator service
+# Description: The Host Orchestrator service provides the signaling
+#              server used by all cuttlefish instances running in this
+#              host as well as orchestration capabilities.
+### END INIT INFO
+#
+# Copyright (C) 2021 The Android Open Source Project
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Make sure calls to this script get redirected to systemctl when
+# using systemd
+
+. /lib/lsb/init-functions
+
+if [ -f /etc/default/cuttlefish-common ]; then
+    . /etc/default/cuttlefish-common
+fi
+orchestrator_port=${orchestrator_port:-1080}
+
+RUN_DIR="/run/cuttlefish"
+ASSET_DIR="/usr/share/cuttlefish-common/host-orchestrator"
+DAEMON="/usr/lib/cuttlefish-common/bin/host-orchestrator"
+PIDFILE="${RUN_DIR}"/host-orchestrator.pid
+
+start() {
+  mkdir -p "${RUN_DIR}"
+  chown _cutf-operator:cvdnetwork "${RUN_DIR}"
+  chmod 775 "${RUN_DIR}"
+
+  ORCHESTRATOR_PORT="${orchestrator_port}" \
+  ORCHESTRATOR_SOCKET_PATH="${RUN_DIR}"/operator \
+  start-stop-daemon --start \
+    --pidfile "${PIDFILE}" \
+    --chuid _cutf-operator:cvdnetwork \
+    --chdir "${ASSET_DIR}" \
+    --background --no-close \
```

Comment:
The man page for `start-stop-daemon` writes

> **`-b`, `--background`**
> ...
> **Warning: `start-stop-daemon`** cannot check the exit status if the process fails to execute for any reason. This is a last resort, and is only meant for programs that either make no sense forking on their own, or where it's not feasible to add the code for them to do this themselves.

Since we control the code of the operator, should we add the forking code so `start-stop-daemon` can tell if the process ran successfully or not?

It looks like this might also require implementing the pidfile functionality internally as well though.

Reply:
The operator is written in go, which doesn't seem to be too friendly to demonizing (no support for fork out of the box) although it can be done through adding a dependency (https://github.com/sevlyar/go-daemon) I read that the preferred way is to use an external tool like demonize.
The main benefit of checking the exit code would be to report whether the process was able to grab the resources it needs (port, socket, etc), which is certainly useful, but I think it's best to add it later as new feature and rely on start-stop-demon for now.